pkg/apiclient/apiclient.go

```diff
@@ -234,6 +235,10 @@ func NewClient(opts *ClientOptions) (Client, error) {
 	}
 	if opts.AuthToken != "" {
 		c.AuthToken = opts.AuthToken
+		_, err := clusterauth.ParseServiceAccountToken(c.AuthToken)
```

Comment:
This function looks specifically designed to parse k8s ServiceAccount tokens. I'm not sure whether it's appropriate for validating Argo CD auth tokens.

What if we just added logic to the CLI to strip leading and trailing whitespace from the token? That seems reasonably safe and would solve the reported issue.

Reply:
```
	if localCfg != nil {
		err = c.refreshAuthToken(localCfg, ctxName, opts.ConfigPath)
		if err != nil {
			return nil, err
		}
	}
```

```
	if err != nil {
		return err
	}
	parser := jwt.NewParser(jwt.WithoutClaimsValidation())
	var claims jwt.RegisteredClaims
	_, _, err = parser.ParseUnverified(configCtx.User.AuthToken, &claims)
	if err != nil {
		return err
	}
	if claims.Valid() == nil {
		// token is still valid
		return nil
	}
```
I think the validating for tokens is the same.

Reply:
I copied the `refreshAuthToken` code when I submitted it for the first time, but it was not elegant.So I imported methods from other packages.

Reply:
@cleverhu honestly, I'd like the space-trimming option much better. It solves the reported issue, and has a very very low risk of accidentally breaking something.

Reply:
Ok，I will change the code.

Reply:
@crenshaw-dev I verified that this is valid, and updated the code.PTAL.